airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3StorageOperations.kt

```diff
@@ -384,6 +459,42 @@ open class S3StorageOperations(
         }
     }
 
+    override fun listExistingObjects(
+        namespace: String?,
+        streamName: String,
+        objectPath: String,
+        pathFormat: String
```

Comment:
So here we'd take an optional min gen id 

If present: filter all keys with metadata.gen_id > min_gen_id?

Reply:
I'm thinking of keeping the StorageOps agnostic of that logic and move that into orchestrating part (OnStart function)

Reply:
actually scratch that. much easier to implement & integration test in StorageOps